### ISSUE TITLE
Add a global fade() function

### ISSUE THREAD

**Issue by tzachyrm** [goldfire/howler.js#1324]
There are global `mute` and `volume` functions. It would be nice to add a global `fade` function as well, that will work on the global volume (sound volumes won't get affected).

My use case is fading in/out all sounds when the browser window becomes visible/invisible.

**Comment by webketje** [comment]
Leaving this here as a fallback reference: 

```js
/**
*  @param howl - the Howler object to use
*  @param targetVolume - number between 0 & 1
*  @param time - time (in seconds) for the fade to complete
**/
var fade = (function() { 
  var intv;

  return function(howl, targetVolume, time) {
     clearInterval(intv);

     var vol = howl.volume();
     var delta = Math.abs(vol - targetVolume);
     var timelapse = delta * time;
     var iterations = time / timelapse;
     var volumeStep = delta / iterations;
     
     if (vol > targetVolume)
       volumeStep = -volumeStep;

     intv = setInterval(function() {
       iterations -= 1;
       vol += volumeStep;
       howl.volume(vol);
       if (iterations === 0) {
         clearInterval(intv);
       }
     }, timelapse);
  }
}());

// usage
function fadeIn(howl) {
  return fade(howl, 1, 3000);
}

function fadeOut(howl) {
  return fade(howl, 0, 3000);
}

```

**Comment by ubershmekel** [comment]
@webketje thanks. Your snippet had a few issues I fixed. Specifically it was taking seconds long steps for me.

```
/**
*  @param howl - the Howler object to use
*  @param targetVolume - number between 0 & 1
*  @param time - time (in ms) for the fade to complete
*  https://github.com/goldfire/howler.js/issues/1324
**/
var fade = (function() { 
  var intv;

  return function(howl, targetVolume, time) {
     clearInterval(intv);

     var vol = howl.volume();
     var delta = targetVolume - vol;
     var intervalMs = 20;
     var iterations = time / intervalMs;
     var volumeStep = delta / iterations;
     
     intv = setInterval(function() {
       iterations -= 1;
       vol += volumeStep;
       console.log("fade step", vol);
       howl.volume(vol);
       if (iterations === 0) {
         clearInterval(intv);
       }
     }, intervalMs);
  }
}());

export function fadeIn() {
  return fade(Howler, 1, 200);
}

export function fadeOut() {
  return fade(Howler, 0, 200);
}
```

**Comment by webketje** [comment]
@ubershmekel You're right I did make a mistake, I passed milliseconds instead of seconds in the `fadeX` functions. Your version is not clean though, intervalMs is dependent on delta & time and cannot be hardcoded if you want an equal distribution of volume change/time, and I doubt you'd want it to fire every 20ms because in this example that means your volume goes from 0 to 1 in 200ms and that is a barely audible fade. I fixed it in the example now, please try again.

**Comment by ubershmekel** [comment]
@webketje while I'm not sure if we should be debugging workarounds on this thread, I will at least explain my reasoning. Even if the overall fade takes multiple seconds, you should update the volume every 10-100 ms to avoid the user noticing the changes.

A fast 200ms fade can be useful to avoid pops in the audio from stopping and stopping sounds. That's what I wanted it for.

Your code snippet has a bug though. Did you test it? For example if I want to go from volume `0` to `1` in `10 seconds` then `timelapse` would be 10, meaning the interval function would be called every 10 seconds. And `iterations` would be `1` meaning there would just be one big change of volume.

I can also tell the math is wrong based on unit arithmetic. Your `timelapse` variable needs to be in `seconds` the way you use it in `setInterval`, but when you calculate it as `delta * time`, the units are `volume * seconds`. I suspect what you really wanted to do was `iterations = 100; timelapse = time / iterations`.

**Comment by webketje** [comment]
@ubershmekel You're right either the number of iterations or the timelapse need to be provided, I appreciate the input. What I really wanted to do was not hardcoding the number of iterations, but keeping the volume change distribution constant over time. My apologies for, as you correctly observed, not testing my previous snippet review
